### ISSUE TITLE
Pen HSL Values

### ISSUE THREAD

**Issue by jsphdvn** [nathan/phosphorus#562]
When you use a 'set pen colour to ()' block, with the **number** input, it treats all values as 0. As phosphorus runs in JS, (I presume) I think you could use HSL values...? Try it. To view the Scratch project, (for proof that I'm not faking this) you can use MegaApuTurkUltra's project downloader, on his page: codepen.io/MegaApuTurkUltra. I was surprised about this glitch.

http://scratch.mit.edu/projects/101518833/
http://phosphorus.github.io/#101518833
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36

**Comment by tjvr** [comment]
I don't think this is true. The `set pen color to (hue)` block has always worked fine for me in phosphorus.

If you're sure this is a bug, maybe you could share a minimal test case demonstrating it?

**Comment by jsphdvn** [comment]
Did you even read my message? It obviously doesn't work... look - it's  
completely red. Then, run the project in Scratch and it's got multiple  
colours.

**Comment by tjvr** [comment]
Please be patient, there are lots of issues to get through. :-)

If you could share the project, that would be a great help.

**Comment by jsphdvn** [comment]
You still didn't read the message. I gave a method of getting the  
project. However, I understand that there are many more important bugs  
to be fixed.

**Comment by tjvr** [comment]
I did see that you suggested using MATU's downloader. However, for bug reports we'd prefer access to the original project.

**Comment by htmlgames** [comment]
Looks like setPenHueTo cannot handle a list index - as a work around could you try modding your
scratch project to use a variable instead of a list directly i.e stick the value of the list index into a variable first and reference that in your set pen color to block?

**Comment by htmlgames** [comment]
Note: fixed in phosphorus.js add line below. This converts scratch NaN trick (0/0) to a js compatible one:

      } else if (block[0] === 'setVar:to:') { /* Data */
	if(block[2] == '/,0,0') block[2] = 'NaN'; // PF fix NaN
        source += varRef(block[1]) + ' = ' + val(block[2]) + ';\n';

**Comment by jsphdvn** [comment]
Great idea. That's some very neat code.
